api/jobs/handlers.py

```diff
@@ -531,7 +539,7 @@ def post(self):
         gear = get_gear(gear_id)
         if gear.get('gear', {}).get('custom', {}).get('flywheel', {}).get('invalid', False):
             self.abort(400, 'Gear marked as invalid, will not run!')
-        validate_gear_config(gear, config_)
+        validate_gear_config(gear, fill_gear_default_values(gear, config_))
```

Comment:
The config defaults will be filled in `Queue.enqueue_job`. Setting it at the beginning is maybe slightly more efficient, but it will still iterate over all of the gear config keys for each batch job that is enqueued. You might have already been aware of this.

Reply:
You're right Megan, but I think he's doing the right thing here. We probably need `fill_gear_default_values` whenever `validate_gear_config` is called.